sqlite_utils/db.py

```diff
@@ -409,7 +414,7 @@ def register(fn):
             fn_name = name or fn.__name__
             arity = len(inspect.signature(fn).parameters)
             if not replace and (fn_name, arity) in self._registered_functions:
-                return fn
+                raise FunctionAlreadyRegistered(f'Already registered function with name "{fn_name}" and identical arity')
```

Comment:
Just realized this represents a backward-incompatible change, so I'm going to back this out of the release.